ユーザー管理機能実装

upamaru0104 opened the pull request:
# What
ユーザー管理機能実装
単体テストコードはまだ作成していません
# Why
ユーザー管理機能実装するため

https://gyazo.com/6c94ebfba9bb3f1bd168b04bf7a03e62
https://gyazo.com/07de3a044451aa7e058765f5ebc2d5e5
https://gyazo.com/62661df042d3bdaeb47e26c40793ccea
https://gyazo.com/60a4ec1e132b957b3b395862295e4be0
https://gyazo.com/6d6d6cb2412015ce7d8662cd8ca373d0
https://gyazo.com/17325d5003e666c5c79004ab98067497

upamaru0104 committed:
devise導入

upamaru0104 committed:
マイグレーションファイル記述

upamaru0104 committed:
userモデルにバリデーションとアソシエーションの記述

upamaru0104 committed:
新規登録のビューファイルの記述の変更

upamaru0104 committed:
ログインビューファイルの記述変更

upamaru0104 committed:
マイグレーションファイル修正

t.string   :reset_password_token
でsyntax errorが表示された
何もミスがないときはその一行前の記述を確認する

upamaru0104 committed:
新規登録ができるようにコントローラーに記載

upamaru0104 committed:
新規登録ビューファイルの修正

パスワードは、半角英数字混合での入力
お名前(全角)は、全角（漢字・ひらがな・カタカナ）での入力
 お名前カナ(全角)は、全角（カタカナ）での入力
について未実装

upamaru0104 committed:
名前フォームと名前の読み仮名フォームに条件を追加

upamaru0104 committed:
パスワードのバリデーション変更

英字と数字の混合でないと登録できないという記述

Kentozzz commented:
レビュー依頼ありがとうございます。
以下2点の確認をお願いします。

### ◇ レビュー依頼は全ての実装を終えてから提出しましょう
恐れ入りますが、ユーザー管理機能はテストまでおこなっていただく必要がございます。
テストに成功することで実装が正しく行えることを確認でき、レビュー提出の準備が完了したことになります。
テストまで終わらせることができたら再度提出をお願いします。

### ◇ Gyazoにタイトルを付けましょう
せっかくGyazoをいただたのたですが、こちらGyazoにタイトルをつけていただく必要があるため再提出をお願いいたします。

```
例）
・必要な情報を適切に入力して「会員登録」ボタンを押すと、ユーザーの新規登録ができる動画
https://gyazo.com/6c94ebfba9bb3f1bd168b04bf7a03e62
```
全てのGyazoにタイトルを付けていただいたら再依頼をお願いいたします。

とはいえ、まずはテストを終わらせる必要があるため、テストが終わり次第Gyazoのタイトルや不足分が無いかのチェックをおこない再依頼をお願いします。
お待ちしております。

upamaru0104 committed:
テストコードに関するgemの導入

upamaru0104 committed:
factorybotに関する記述

upamaru0104 committed:
ユーザーの新規登録に関するテストの記述

upamaru0104 committed:
factorybotの修正

日本語は自動で生成されないため自分で設定する

upamaru0104 committed:
単体テストコードの記述

バリデーションでpresence: trueと設定されていたらcan't be blankというエラーメッセージがデフォルトで設定される
それ以外はすべてデフォルトでin validと表示される
モデルのオプションでエラーメッセージを設定しているときはそれをエラー文で設定する

upamaru0104 commented:
#What
ユーザー管理機能実装
#Why
ユーザー管理機能実装するため

必要な情報を適切に入力して「会員登録」ボタンを押すと、ユーザーの新規登録ができる動画
https://gyazo.com/9d4d1fb34fc0d0dd9be1347a10218367
入力に問題のある状態で「会員登録」ボタンが押された場合、ユーザーの新規登録はできず、エラーメッセージが出力される動画
https://gyazo.com/5f8306b82f2b93a2a5382882d7978897
必要な情報を適切に入力して「ログイン」ボタンを押すと、ユーザーのログインができる動画
https://gyazo.com/f04974c7af9e9659f6467dfe08013d84
入力に問題のある状態で「ログイン」ボタンが押された場合、ユーザーのログインはできず、エラーメッセージが出力される動画
https://gyazo.com/58f4ef60d8455656b349a3fdec5b8243
ログアウトに成功する動画
https://gyazo.com/6dc1a0ad1f05b04ef87579bf5c9716e4
 テストに成功する画像
https://gyazo.com/ca7e74b520b639cd01bd1dcea5f20885

ShotaOtake commented on app/models/user.rb:
```diff
@@ -0,0 +1,22 @@
+  validates :nickname,           presence: true
+  validates :birthday,           presence: true
+
+  has_many :items
```
◎内容をコメントアウトしましょう

```
  has_many :items
  has_many :purchase_records
```

他機能の内容であるため

ShotaOtake commented on app/views/devise/registrations/new.html.erb:
```diff
@@ -1,7 +1,7 @@
 <%= render "shared/second-header"%>
 
 <%# 「モデル名」にはUserモデルであれば@userを渡しましょう。「新規登録機能へのパス」は、devise導入後にrails routesを実行してdevise/registrations#createへのパスを確認し、記載してください。 %>
```
実装要件のコメントアウトですが、不要であれば【すべて】削除しましょう！

```
・原則、デフォルトのコメントアウトは、実装後に削除頂きます。
(あくまで実装時のヒントであるため)

・もし別途コメントアウトが必要な際は、第三者が分かる形で**「残す目的」や「必要な理由」**を、コメントとして書いておくことをおすすめ致します。
```

ShotaOtake commented on app/views/shared/_header.html.erb:
```diff
@@ -16,10 +16,13 @@
     </ul>
     <ul class='lists-right'>
       <%# deviseを導入できたら、ログインの有無で表示が変わるように分岐しましょう%>
```
実装要件のコメントアウトですが、不要であれば【すべて】削除しましょう！

```
・原則、デフォルトのコメントアウトは、実装後に削除頂きます。
(あくまで実装時のヒントであるため)

・もし別途コメントアウトが必要な際は、第三者が分かる形で**「残す目的」や「必要な理由」**を、コメントとして書いておくことをおすすめ致します。
```

ShotaOtake commented on spec/factories/users.rb:
```diff
@@ -0,0 +1,13 @@
+  factory :user do
+    nickname              {Faker::Name.initials(number: 2)}
+    email                 {Faker::Internet.free_email}
+    password              {Faker::Internet.password(min_length: 6, mix_case: true, special_characters: false)}
```
■パスワードのFactoryの内容を修正しましょう！

```
password {Faker::Internet.password(min_length: 6, mix_case: true, special_characters: false)}
```

現状、Fakerでの生成となりますが、必ずしも半角英数混合になるかどうかが不明となる記述です。
(Fakerのみの場合、英字のみのパスワードが生成されてしまうことがある)

確実に半角英数混合の値を生成するため、文字列を組み合わせることで期待する値を作るような内容に修正していきましょう！

ShotaOtake commented on spec/models/user_spec.rb:
```diff
@@ -0,0 +1,116 @@
+        @user.last_name = 'abc'
+        @user.valid?
+        expect(@user.errors.full_messages).to include('Last name is invalid. Input full-width characters.')
+      end
```
■テストの要件を追加しましょう！

現状、下記の要件が満たされていない状況でございます。

```

・英字のみのパスワードでは登録できない
・数字のみのパスワードでは登録できない
・全角文字を含むパスワードでは登録できない

※パスワードにつきましては上記それぞれの項目に分けてexampleの生成をお願い致します。

```

上記を全て満たせるようexampleを追加しましょう！

ShotaOtake reviewed (commented):
レビュー依頼を頂きありがとうございます！
追加で修正依頼のコメントをさせて頂きましたので、ご確認の上修正をお願い致します。

完成まであと一歩です！
引き続き実装頑張っていきましょう！

```
※テストコードにて修正箇所のございます際は、お手数ですが、最新の実行結果の画像も改めて添付頂きますようお願い致します。
```

upamaru0104 committed:
コメントアウトの削除

upamaru0104 committed:
コメントアウト追加

upamaru0104 committed:
パスワード修正

upamaru0104 committed:
テストの追加

upamaru0104 commented:
修正しました
passwordが数字だけでは登録できないというのと
数字のみのパスワードでは登録できないは
違うということですか
テストに成功する画像
https://gyazo.com/0ca00c930513715f9bcd3b46dfe7b108

ShotaOtake reviewed (commented):
レビュー依頼を頂きありがとうございます！
お送り頂いた内容を確認致しました。

今回【LGTM】とさせて頂きます。
実装お疲れ様でした！

```
※現在のブランチはマージ後、ローカルのmasterブランチに忘れずにプルしましょう！
その後、新しいブランチを作成し次の実装に取り組んでいただけますと幸いです。
```

引き続き実装頑張っていきましょう！
応援しております！

upamaru0104 committed:
修正
